src/mjolnir/shortcutbuilder.cc

```diff
@@ -315,11 +294,30 @@ uint32_t ConnectEdges(GraphReader& reader,
                       GraphId& endnode,
                       uint32_t& opp_local_idx,
                       uint32_t& restrictions,
-                      float& average_density) {
+                      float& average_density,
+                      float& total_duration,
+                      float& total_truck_duration) {
   // Get the tile and directed edge.
   const GraphTile* tile = reader.GetGraphTile(startnode);
   const DirectedEdge* directededge = tile->directededge(edgeid);
 
+  // Add edge and turn duration for car
+  auto const nodeinfo = tile->node(startnode);
+  auto const turn_duration = OSRMCarTurnDuration(directededge, nodeinfo, opp_local_idx);
+  total_duration += turn_duration;
+  auto const speed = directededge->speed();
+  // TODO: check speed for zero
```

Comment:
In doubt if it's needed 🤔

Reply:
add an assert here that its not 0, should be good enough